sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -259,6 +239,13 @@ RetT *getOrBuild(KernelProgramCache &KPCache, KeyT &&CacheKey,
 
 static bool isDeviceBinaryTypeSupported(const context &C,
                                         RT::PiDeviceBinaryType Format) {
+  const backend ContextBackend =
+      detail::getSyclObjImpl(C)->getPlugin().getBackend();
+
+  // The CUDA backend cannot use SPIRV
+  if (ContextBackend == backend::cuda && Format == PI_DEVICE_BINARY_TYPE_SPIRV)
+    return false;
```

Comment:
Both PTX and CUBIN are currently PI_DEVICE_BINARY_TYPE_NATIVE.
We originally implemented `piProgramCreateWithSource`  to consume PTX or CUBIN, but that is actually incorrect, and `piProgramCreateWithBinary` is the correct choice for both.  The current code is "plumbed" so that it calls `piProgramCreateWithSource` but should do `piProgramCreateWithBinary`. An incoming patch should fix that.

Reply:
ok